packages/contracts/contracts/L1/rollup/OVM_CanonicalTransactionChain.sol

```diff
@@ -273,24 +275,31 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
             "Transaction gas limit too low to enqueue."
         );
 
-        // We need to consume some amount of L1 gas in order to rate limit transactions going into
-        // L2. However, L2 is cheaper than L1 so we only need to burn some small proportion of the
-        // provided L1 gas.
-        uint256 gasToConsume = _gasLimit/L2_GAS_DISCOUNT_DIVISOR;
-        uint256 startingGas = gasleft();
-
-        // Although this check is not necessary (burn below will run out of gas if not true), it
-        // gives the user an explicit reason as to why the enqueue attempt failed.
-        require(
-            startingGas > gasToConsume,
-            "Insufficient gas for L2 rate limiting burn."
-        );
+        // Transactions submitted to the queue are not routed through the Sequencer Entrypoint,
+        // therefore they cannot be forced to pay a fee to the Sequencer. So we need to prevent spam
+        // attacks by ensuring that the cost of enqueueing a transaction from L1 to L2 is not
+        // underpriced.
+        // Therefore, we define 'ENQUEUE_L2_GAS_PREPAID' as a threshold. If the _gasLimit for the
+        // enqueued transaction is above this threshold, then we 'charge' to user by burning
+        // additional L1 gas. Since gas is cheaper on L2 than L1, we only need to burn a fraction of
```

Comment:
Gas is cheaper on L2 because the minimum hardware requirement is higher, not sure if that is worth being explicit about but without that context that statement could make no sense

Reply:
I chose to leave that out, feels tangential.